Http/Controllers/Admin/MenuItemController.php

```diff
@@ -70,12 +75,85 @@ public function destroy(Menu $menu, Menuitem $menuItem)
     }
 
     /**
+     * @param Menu, $menuItemId
+     * @return array
+     */
+    private function getMenuSelect($menu, $menuItemId = null)
+    {
+        $menus = $menu->all();
+
+        $menuSelect = array();
+
+        foreach ($menus as $menuEntity) {
+            $menuSelect[$menuEntity['name']] = $menuEntity->menuitems()->get();
+        }
+
+        return $menuSelect;
+    }
+
+    /**
      * @param  Menu                                    $menu
      * @param  \Illuminate\Foundation\Http\FormRequest $request
      * @return array
      */
     private function addMenuId(Menu $menu, FormRequest $request)
     {
-        return array_merge($request->all(), ['menu_id' => $menu->id]);
+        $data = $request->all();
+        foreach (\LaravelLocalization::getSupportedLanguagesKeys() as $lang) {
+            $uri = $data[$lang]['uri'];
+            $data[$lang]['uri'] = ! empty($data['link_type'] == 'internal') ? $uri : $this->getUri($data['page_id'], $lang);
+        }
+        unset($data['parent_id']); // Just for the moment until Selectbox is ready
+
+        return array_merge($data, ['menu_id' => $menu->id]);
+    }
+
+    /**
+     * Get uri
+     *
+     * @param $item
+     * @return string
+     */
+    private function getUri($pageId, $lang)
+    {
+        $linkPathArray = array();
+
+        array_push($linkPathArray, $this->getPageSlug($pageId, $lang));
+
+        $currentItem = Menuitem::where('page_id', '=', $pageId)->first();
+
+        $hasParentItem = false;
+
+        while ($hasParentItem) {
+            $parentItemId = isset($parentItem) ? $parentItem->parent_id : $currentItem->parent_id;
+
+            $parentItem = Menuitem::where('id', '=', $parentItemId)->first();
+
+            if ($parentItem->is_root != true) {
+                if (!empty($parentItem->page_id)) {
+                    array_push($linkPathArray, $this->getPageSlug($parentItem->page_id, $lang));
+                } else {
+                    $parentUri = ! is_null($parentItem->uri) ? $parentItem->uri . '/' . $linkPathArray : $linkPathArray;
+                    array_push($linkPathArray, $parentUri);
+                }
+            }
+
+            $hasParentItem = ! is_null($parentItem->parent_id) ? true : false;
+        }
+
+        $parentLinkPath = implode('/', array_reverse($linkPathArray));
+
+        return $parentLinkPath;
+    }
+
+    /**
+     * Get page slug
+     *
+     * @params $pageId, $lang
+     * @return string
+     */
+    private function getPageSlug($pageId, $lang)
+    {
+        return Page::where('id', '=', $pageId)->first()->translate($lang)->slug;
```

Comment:
Use the page repository. Model methods should never be used outside the repository.